### PR TITLE
Getting disk usage details

### PR DESCRIPTION
**Pull request by baldhead**
This may help https://github.com/plesk/api-php-lib/issues/24

### CODE REVIEW

File: src/Api/Operator/Webspace.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Copyright 1999-2019. Plesk International GmbH.
 
 namespace PleskX\Api\Operator;
+
```
Comment (mrsombre) [inline src/Api/Operator/Webspace.php]: Remove white-space change

File: src/Api/Operator/Webspace.php [hunk 2]
```diff
@@ -34,7 +35,7 @@ public function getPhysicalHostingDescriptor()
     public function create(array $properties, array $hostingProperties = null, $planName = null)
     {
         $packet = $this->_client->getPacket();
-        $info = $packet->addChild($this->_wrapperTag)->addChild('add');
+        $info   = $packet->addChild($this->_wrapperTag)->addChild('add');
```
Comment (mrsombre) [inline src/Api/Operator/Webspace.php]: Remove white-space change

File: src/Api/Operator/Webspace.php [hunk 3]
```diff
@@ -91,4 +92,15 @@ public function getAll()
...
+     * @param string $field
+     * @param integer|string $value
+     * @return Struct\DiskUsage
+     */
+    public function getDiskUsage($field, $value)
+    {
+        $items = $this->_getItems(Struct\DiskUsage::class, 'disk_usage', $field, $value);
+        return reset($items);
+    }
+
```
Comment (mrsombre) [inline src/Api/Operator/Webspace.php]: Remove white-space change

**Comment by baldhead** [comment]
Formatting issues should be resolved now.

File: src/Api/Struct/Webspace/DiskUsage.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+// Copyright 1999-2019. Plesk International GmbH.
+// Author: Frederic Leclercq
+
+namespace PleskX\Api\Struct\Webspace;
+
+class DiskUsage extends \PleskX\Api\Struct
+{
+    /** @var string */
+    public $name;
+
```
Comment (mrsombre) [inline src/Api/Struct/Webspace/DiskUsage.php]: Seems name and guid not returned for disk_usage according to reference
https://docs.plesk.com/en-US/onyx/api-rpc/about-xml-api/reference/managing-sites-domains/getting-information-about-sites.66583/
Lets remove this

File: tests/WebspaceTest.php [hunk 5]
```diff
@@ -18,6 +18,13 @@ public function testGetLimitDescriptor()
         $this->assertNotEmpty($descriptor->limits);
     }
 
+    public function testGetDiskUsage()
+    {
+        $diskusage = static::$_client->webspace()->getDiskUsage();
```
[the remark this answers is not in the document]
Comment (mrsombre) [reply tests/WebspaceTest.php]: Consider to change like this
```
public function testGetDiskUsage()
    {
        $webspace = static::_createWebspace();
        $diskusage = static::$_client->webspace()->getDiskUsage('id', $webspace->id);

        $this->assertObjectHasAttribute('httpdocs', $diskusage);
    }
```

[the remark this answers is not in the document]
Comment (mrsombre) [reply tests/WebspaceTest.php]: `assertNotEmpty` is not acceptable because can be zero for new subscriptions

File: tests/WebspaceTest.php [hunk 6]
```diff
@@ -18,6 +18,13 @@ public function testGetLimitDescriptor()
         $this->assertNotEmpty($descriptor->limits);
     }
 
+    public function testGetDiskUsage()
```
Comment (mrsombre) [inline tests/WebspaceTest.php]: You can run tests on demo sites
https://support.plesk.com/hc/en-us/articles/115001407434-How-to-access-Plesk-demo-online-servers
Use `phpunit.xml` for tests configuration

File: tests/WebspaceTest.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (baldhead) [reply tests/WebspaceTest.php]: You're correct, I've changed this...

**Pull request merged**